src/pages/[type].tsx

```diff
@@ -0,0 +1,23 @@
+import { useRouter } from "next/router";
+import { AbsoluteCenter, Container, Spinner } from "@chakra-ui/react";
+
+import CategoryPage from "~/components/CategoryPage";
+import type { ProductType } from "~/types/product";
+
+const ProductTypePage = () => {
+  const {
+    query: { type: product },
+  } = useRouter();
+
+  return product ? (
+    <CategoryPage product={product as ProductType} />
+  ) : (
+    <Container minH={"calc(100vh - 98px)"}>
```

Comment:
is 98px a constant? useLayoutEffect might be a better option for calculating navbar height

Reply:
i think the navbar height in total is hardcoded at 98px, but this should probably not be in case that changes. i'll look into useLayoutEffect